includes/Common-Include.hs

```diff
@@ -82,3 +102,51 @@ hSupportsANSI h = (&&) <$> hIsTerminalDevice h <*> (not <$> isDumb)
   where
     -- cannot use lookupEnv since it only appeared in GHC 7.6
     isDumb = maybe False (== "dumb") . lookup "TERM" <$> getEnvironment
+
+-- | Parses the characters emitted by 'reportCursorPosition' into the console
+-- input stream. Returns the cursor row and column as a tuple.
+--
+-- For example, if the characters emitted by 'reportCursorPosition' are in
+-- 'String' @input@ then the parser could be applied like this:
+--
+-- > let result = readP_to_S cursorPosition input
+-- > case result of
+-- >     [] -> putStrLn $ "Error: could not parse " ++ show input
+-- >     [((row, column), _)] -> putStrLn $ "The cursor was at row " ++ show row
+-- >         ++ " and column" ++ show column ++ "."
+-- >     (_:_) -> putStrLn $ "Error: parse not unique"
+--
+cursorPosition :: ReadP (Int, Int)
+cursorPosition = do
+    void $ char '\ESC'
+    void $ char '['
+    row <- decimal -- A non-negative whole decimal number
+    void $ char ';'
+    col <- decimal -- A non-negative whole decimal number
+    void $ char 'R'
+    return (read row, read col)
+  where
+    digit = satisfy isDigit
+    decimal = many1 digit
+
+-- | Attempts to get the reported cursor position data from the console input
+-- stream. The function is intended to be called immediately after
+-- 'reportCursorPosition' (or related functions) have caused characters to be
+-- emitted into the stream.
+--
+-- For example, on a Unix-like operating system:
+--
+-- > hSetBuffering stdin NoBuffering -- set no buffering (the contents of the
+-- >                                 -- buffer will be discarded, so this needs
+-- >                                 -- to be done before the cursor positon is
+-- >                                 -- emitted)
+-- > reportCursorPosition
+-- > hFlush stdout -- ensure the report cursor position code is sent to the
+-- >               -- operating system
+-- > input <- getReportedCursorPosition
+--
+getReportedCursorPosition :: IO String
+
+-- | Attempts to get the reported cursor position, combining the functions
+-- 'reportCursorPosition', 'getReportedCursorPosition' and 'cursorPosition'.
+getCursorPosition :: IO (Maybe (Int, Int))
```

Comment:
Could you add a couple of words on when this may return `Nothing`?

Reply:
Documentation added to explain `Nothing` and also to warn about non-applicability of `getReportedCursorPosition` and `getCursorPosition` in the case of non-Win32 consoles on Windows.